src/lib/settings/settings.py

```diff
@@ -0,0 +1,214 @@
+        return f"arn:aws:iam::{account_id}:role/role-salmon-cross-account-extract-metrics-dev"
+
+    def _process_monitoring_groups(self):
+        # TODO: add wildcards replacement for all the resource types (glue, lambda, etc.)
+        pass
+
+    # Get raw settings by file name
+    def get_raw_settings(self, file_name: str) -> dict:
+        """Get raw settings by file name"""
+        return self._raw_settings[file_name]
+
+    # CDK methods
+    def get_monitored_account_ids(self) -> List[str]:
+        """Get monitored account_ids"""
+        return [
+            m_env["account_id"]
+            for m_env in self.processed_settings[SettingFileNames.GENERAL].get(
+                "monitored_environments", []
+            )
+        ]
+
+    def get_metrics_collection_interval_min(self) -> int:
+        """Get metrics_collection_interval_min"""
+        return self.processed_settings[SettingFileNames.GENERAL]["tooling_environment"][
+            "metrics_collection_interval_min"
+        ]
+
+    # Lambda methods
+    def get_monitored_environment_name(self, account_id: str, region: str) -> str:
+        """Get monitored environment name by account_id and region."""
+        for m_env in self.processed_settings[SettingFileNames.GENERAL].get(
+            "monitored_environments", []
+        ):
+            if m_env["account_id"] == account_id and m_env["region"] == region:
+                return m_env["name"]
+        return None
+
+    def get_monitoring_groups(self, resources: List[str]) -> List[str]:
+        """Get monitoring groups by resources list."""
+        matched_groups = set()  # Prevent duplicates
+
+        for group in self.monitoring_groups.get("monitoring_groups", []):
+            resource_groups = []
+            for monitored_resource in SettingConfigs.RESOURCE_TYPES:
+                resource_groups += group.get(monitored_resource, [])
+
+            for resource in resources:
+                matched_groups.update(
+                    group["group_name"]
+                    for res in resource_groups
+                    # TODO: replace fnmatch with == when _process_monitoring_groups() implemented
+                    if res["name"] and fnmatch(resource, res.get("name"))
+                )
+
+        return list(matched_groups)
+
+    def get_recipients(
+        self, monitoring_groups: List[str], notification_type: str
+    ) -> List[dict]:
+        """Get recipients by monitoring groups."""
+        matched_recipients = []
+
+        for recipient in self.recipients.get("recipients", []):
+            subscriptions = recipient.get("subscriptions", [])
+            for subscription in subscriptions:
+                for monitoring_group in monitoring_groups:
+                    if subscription.get("monitoring_group") == monitoring_group:
+                        if (
+                            notification_type == "alert" and subscription.get("alerts")
```

Comment:
Maybe it would be better to create a class with constants (enum) for that? Would be more user-friendly

Reply:
Good notice! Done